src/App.tsx

```diff
@@ -14,11 +16,98 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[];
+};
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoods: [],
+  };
+
+  alreadyHasGoods = (good: string) => {
```

Comment:
```suggestion
  isGoodAdded = (good: string) => {
```
функции, которые возвращают boolean - должны начинаться с is/was/has/should и так далее, чтобы это было понятно по названию